docs/src/man/doctests.md

```diff
@@ -389,3 +389,28 @@ Alternatively, you can also pass the `doctest = :fix` keyword to [`makedocs`](@r
 Doctesting can be disabled by setting the [`makedocs`](@ref) keyword `doctest = false`.
 This should only be done when initially laying out the structure of a package's
 documentation, after which it's encouraged to always run doctests when building docs.
+
+## Testing README.md
+
+The easiest way to also run doctests on the README.md file is to add the readme to the
+docs of the module:
+
+```julia
+module MyModule
+
+function _update_module_doc()
+    path = joinpath(pkgdir(MyModule), "README.md")
+    text = read(path, String)
+    @doc text MyModule
+end
+_update_module_doc()
+
+[...]
+
+end # module
+```
+
+Now, when users type `?MyModule` in the REPL, they'll see the contents of the readme
+file, and doctest blocks in the readme file will be tested. Be sure to call 
+`MyModule._update_module_doc()` before running the doctests.
```

Comment:
I don't understand why you would need to call the `_update_module_doc()` manually. Isn't the docstring be attached to the module at precompile time?

Reply:
It is for situations where the user doesn't close the REPL in between running doctests.

Reply:
To verify this, try to run `Pkg.test()` again after changing the README.md and not restarting the REPL. The tests will then use the old doctests (Julia 1.6).